### ISSUE TITLE
Custom Backbone

### ISSUE THREAD

**Issue by cankocagil** [facebookresearch/detr#378]
Hello, I am very impressed with your work. As so, I want to train your model from scratch with different 'custom' backbone. I want to test my custom backbone with your detection strategy for my research. Hence, I want to integrate my backbone architecture to your detection system. I understand that I need to freeze BN layers, so that's ok.  So, what should be the dimensions of the output of the backbone, and how can I integrate your system? I integrated my custom backbone to your demo detr model that you provide for inferencing with little changes. But, I could not find a way to integrate the backbone to your real model. I will be appreciated if you can provide tips & information, or anything that can help me. Best!

**Comment by alcinos** [comment]
Hello @cankocagil 

I think the simplest way to use a custom backbone with minimal changes to the code would be to define it as closely as possible to the resnet general structure, ie with 4 stages at different resolution. What you put in each of the stages doesn't matter, as long as the resolutions are correct. I encourage you to look at the torchvision reference implementation for resnet.

Best of luck

**Comment by SISTMrL** [comment]
have you tried to replace the backbone of detr with swin transformer, thanks

**Comment by cankocagil** [comment]
Yes I succeed @SISTMrL. But, I tried for demo detr model. If you are interested, I can share the code.

**Comment by SISTMrL** [comment]
thanks very much, I'd appreciate it if you could share your code！！！！！！

**Comment by SISTMrL** [comment]
> Yes I succeed @SISTMrL. But, I tried for demo detr model. If you are interested, I can share the code.
thanks very much, I'd appreciate it if you could share your code！！！！！！

**Comment by SISTMrL** [comment]
@cankocagil thanks very much, I'd appreciate it if you could share your code！！！！！！

**Comment by cqg1997** [comment]
@cankocagil  Could you mind sharing your code and  the best checkpoint file. It would be very helpful for me because the trainning schedule is too long. :)

**Comment by cankocagil** [comment]
@cqg1997 Hey! Please send me an email (that can be found at my repo) so that I can share my code in private. I did not further train Swin+Detr after testing whether it works or not (converging or not) because as you mention trainning schedule is too long.

**Comment by cqg1997** [comment]
@cankocagil   Thanks very much for your help. I have sent an email to you.

**Comment by vasanthi457** [comment]
Hello,  I want to train your model from scratch with different 'custom' backbone. I want to test my custom backbone with your detection strategy for my research.   But, I could not find a way to integrate the backbone to your real model. please  can you provide tips & information, or anything that can help me. thanking you...

**Comment by Rares926** [comment]
Hi everyone is there any chance that somebody managed to change the backbone and is willing to share the code with me, or at least explain which would be the steps to realize this.

**Comment by arnozxn** [comment]
> 嗨，大家好，是否有可能有人设法更改了主干并愿意与我共享代码，或者至少解释哪些步骤是实现这一点的步骤。

I did it,but i got a problem when i train
